### ISSUE TITLE
What the heck is this example doing?!?!

### ISSUE THREAD

**Issue by hugotoledo** [MicrosoftDocs/powerbi-docs#2426]
I'm guessing the third line in the DAX example is a mistake. If it isn't, I'd truly love to know what I'm missing.

IF(
    USERNAME() = "Worker",
  __[Type] = "Internal",__
    IF(
        USERNAME() = "Manager",
        TRUE(),
        FALSE()
    )
)

Why would you use such an arcane example? I don't know what the possible values of [Type] are so how do I know what [Type] = "Internal" will resolve to? Is it True? Is it False?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8e49a178-5b5e-ae62-b12c-be846eb6ce40
* Version Independent ID: 8372b4a0-75e5-dc84-cdbf-e1c2d76331a7
* Content: [Row-level security (RLS) guidance in Power BI Desktop - Power BI](https://docs.microsoft.com/en-us/power-bi/guidance/rls-guidance)
* Content Source: [powerbi-docs/guidance/rls-guidance.md](https://github.com/MicrosoftDocs/powerbi-docs/blob/live/powerbi-docs/guidance/rls-guidance.md)
* Service: **powerbi**
* Sub-service: **powerbi-desktop**
* GitHub Login: @peter-myers
* Microsoft Alias: **v-pemyer**

**Comment by kfollis** [comment]
Thanks for the feedback, @hugotoledo. I think that's a reference to the imaginary data, but I'm adding the author to provide clarification.

**Comment by peter-myers** [comment]
Thanks @hugotoledo - we will take your feedback into consideration. #please-close

**Comment by DaniilMaslyuk** [comment]
@hugotoledo just in case you still would like to understand what's going on -- the expression can be re-written in the following way:

    IF(
        USERNAME() = "Worker",
        IF(
            [Type] = "Internal",
            TRUE(),
            FALSE()
        ),
        IF(
            USERNAME() = "Manager",
            TRUE(),
            FALSE()
        )
    )

So the logic becomes:
Is the user a **Worker**?
1. If the user is a **Worker**, is the type **Internal**?
1.1. If the type is **Internal**, the show the row
1.2. If the type is not **Internal**, don't show the row
2. If the user is not a **Worker**, is the user a **Manager**?
2.1. If the user is a **Manager**, show the row
2.2. Is the user is not a **Manager**, don't show the row

So in short, `[Type] = "Internal"` is the same as `IF([Type] = "Internal", TRUE(), FALSE())`

**Comment by hugotoledo** [comment]
I'm sorry I'd not written back to thank you for this earlier. It very much helped me understand a syntax feature I'd utterly missed. It seems obvious in retrospect. Thanks a million!
